File: src/main/java/main/data/persistence/TeamRepository.java
package main.data.persistence;

import main.data.Division;
import main.data.Team;
import org.springframework.data.domain.Sort;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

@Repository
@Transactional
public interface TeamRepository extends PagingAndSortingRepository<Team, Integer> {
    Team findById(int id);
    Team findByUsername(String username);
    Iterable<Team> findBySchool(String school);
    Iterable<Team> findByDivision(Division division);
    Iterable<Team> findByDivision(Division division, Sort sort);
}

File: src/main/kotlin/main/data/persistence/TeamRepository.kt
package main.data.persistence

import main.data.Division
import main.data.Team
import org.springframework.data.domain.Sort
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional
import java.util.*

@Repository
@Transactional
interface TeamRepository : PagingAndSortingRepository<Team, Int> {
    fun findByUsername(username: String): Optional<Team>
    fun findBySchool(school: String): Iterable<Team>
    fun findByDivision(division: Division): Iterable<Team>
    fun findByDivision(division: Division, sort: Sort): Iterable<Team>
}
